subsys/testsuite/ztest/src/ztest_shell.c

```diff
@@ -0,0 +1,416 @@
+	"                         enable / disable colored output. the default is off\n"
+	"--" PREFIX "print_time=(0|1)\n"
+	"                         print time duration for each test (defaults to 1)\n"
+	"--" PREFIX "output=(json|xml)[:DIRECTORY_PATH/|:FILE_PATH]\n"
+	"                         generate a json or xml report in the given directory or with\n"
+	"                         the given filename. The default is not to generate a report\n"
+	"\n"
+	"See https://pastebin.com/GtBg0aHZ for details\n";
+
+static int ztest_shell_parse_opts(const struct shell *sh, int argc, char *const *argv,
+				  struct ztest_shell_state *opts)
+{
+	int ret;
+	int option_index = 1;
+
+	while ((ret = getopt_long(argc, argv, "", ztest_shell_longopts, &option_index)) != -1) {
+		switch (ret) {
+		case 'h':
+			opts->help = true;
+			break;
+		case 'l':
+			opts->list_tests = true;
+			break;
+		case 'f':
+			opts->filter = optarg;
+			break;
+		case 'd':
+			opts->disabled = true;
+			shell_warn(sh, "running disabled tests is not yet implemented");
+			break;
+		case 'r':
+			opts->repeat = atoi(optarg);
+			break;
+		case 'u':
+			opts->shuffle = true;
+			shell_warn(sh, "shuffle is not yet implemented");
+			break;
+		case 's':
+			opts->seed = atoi(optarg);
+			shell_warn(sh, "random_seed is not yet implemented");
+			break;
+		case 'c':
+			opts->color = strcmp(optarg, "no") != 0;
+			shell_warn(sh, "color is not yet implemented");
+			break;
+		case 't':
+			opts->time = (bool)atoi(optarg);
+			shell_warn(sh, "print_time is not yet implemented");
+			break;
+		case 'o':
+			opts->output = optarg;
+			shell_warn(sh, "writing output to file is not yet implemented");
+			break;
+		case '?':
+		default:
+			return -EINVAL;
+		}
+	}
+
+	return 0;
+}
+
+static int ztest_shell_verify_opts(const struct shell *sh, struct ztest_shell_state *opts)
+{
+	ARG_UNUSED(sh);
+	ARG_UNUSED(opts);
+
+	return 0;
+}
+
+static int cmd_ztest(const struct shell *sh, size_t argc, char *const *argv)
+{
+	int ret;
+
+	_state = (struct ztest_shell_state){
+		.repeat = 1,
+		.time = true,
+		.sh = sh,
+	};
+
+	ret = ztest_shell_parse_opts(sh, argc, argv, &_state);
+	if (ret < 0) {
+		return ret;
+	}
+
+	ret = ztest_shell_verify_opts(sh, &_state);
+	if (ret < 0) {
+		return ret;
+	}
+
+	return ztest_shell_execute(sh, &_state);
+}
+
+SHELL_CMD_ARG_REGISTER(ztest, NULL, cmd_ztest_help, cmd_ztest, 0, N_ARGS);
+
+bool z_ztest_should_test_run(const char *suite, const char *test)
+{
+	char name[CONFIG_ZTEST_SHELL_NAME_SIZE_MAX];
+	size_t name_size = strlen(suite) + 1 /* '.' */ + strlen(test) + 1 /* '\0' */;
+
+	if (name_size > CONFIG_ZTEST_SHELL_NAME_SIZE_MAX) {
+		printk("Warning: CONFIG_ZTEST_SHELL_NAME_SIZE_MAX is not large enough for\n");
+		printk("Warning: %s.%s\n", suite, test);
+		printk("Warning: Needed: %zu\n", name_size);
+		printk("Warning: Actual: %zu\n", (size_t)CONFIG_ZTEST_SHELL_NAME_SIZE_MAX);
+		return false;
+	}
+
+	snprintf(name, ARRAY_SIZE(name), "%s.%s", suite, test);
```

Comment:
I think this should be a `sizeof` rather than `ARRAY_SIZE` since snprintf writes in bytes and not "elements"

Reply:
yeah, that works too. I think they're equivalent more or less here.

Reply:
yeah I just think it's inaccurate in this case

Reply:
The only downside of using sizeof is that it works for pointers too, while array size only works for arrays.
Anyways. Fixed!